### PR TITLE
At 47 implement all reports

### PR DESCRIPTION
**Pull request by valerymolinalika**
Here is the PR. Now Johan can stop complaining and making my life miserable. Thanks and kisses.

### CODE REVIEW

File: telegram/commands/report/all.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+// GenerateAllReports generates all reports and send it in txt file
+func GenerateAllReports(bot *shared.Bot, userName, content string, chatID int64) error {
+	reports := ""
+	filePath := "all_reports.txt"
+
+	for _, fn := range reportsFunctions {
+		report, err := fn(bot, userName, chatID)
+		if err != nil {
+			_ = shared.SendMessage(bot, chatID, "f")
```
Comment (bescobarm) [inline telegram/commands/report/all.go:26]: define a better error message to this scenario

File: telegram/commands/report/run.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+	Lunes: %s
+	Martes: %s
+	Miércoles: %s
+	Jueves: %s
+	Viernes: %s
+	Sábado: %s
+	Domingo: %s
+
+	Si no querés las pantorrillas lesionadas de johan ponete a correr un poquito más 🤢
```
Comment (johanflorez219) [inline telegram/commands/report/run.go:21]: jasjasjs seria correr menos no? sapa perra

File: telegram/commands/report/run.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	`
+
+	labelTookRun = "sisas"
+)
+
+// SendRunReport sends the run report
+func SendRunReport(bot *shared.Bot, userName, content string, chatID int64) error {
+	kr, err := GenerateRunReport(bot, userName, chatID)
```
Comment (johanflorez219) [inline telegram/commands/report/run.go:29]: what's kr?

File: telegram/commands/report/run.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+		if err != nil {
+			return "", shared.SendMessage(bot, chatID, fmt.Sprintf(shared.ErrSendMessage, err.Error()))
+		}
+
+		day := createdAt.Weekday()
+
+		tookRun[day] = labelTookRun
+	}
+
+	report := fmt.Sprintf(reportRunMessage, userName, tookRun[time.Monday], tookRun[time.Tuesday], tookRun[time.Wednesday], tookRun[time.Thursday], tookRun[time.Friday], tookRun[time.Saturday], tookRun[time.Sunday])
```
Comment (johanflorez219) [inline telegram/commands/report/run.go:67]: what a piece of shit

you could sort the map and send it

File: telegram/commands/report/shower.go [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+Domingo: %s
+
+Última racha:
+Bañándose: %d %s 🚿
+Sin bañarse: %d %s 🤢
+
+Andá bañate pa' que estos números mejoren, cochin@ de 💩💩`
+
+	oneDayLabel        = "día"
+	textSisasStreak    = "días"
```
Comment (johanflorez219) [inline telegram/commands/report/shower.go:29]: sisas = dias? what the heck?

File: telegram/commands/report/tooth.go [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+	Sábado: %d
+	Domingo: %d
+
+	Con estos números se avecina una caries ni la hp, ponete juicios@ pues 🤢
+	`
+)
+
+// SendToothReport sends the tooth report
+func SendToothReport(bot *shared.Bot, userName, content string, chatID int64) error {
+	wr, err := GenerateToothReport(bot, userName, chatID)
```
Comment (johanflorez219) [inline telegram/commands/report/tooth.go:28]: whats wr?

**Comment by johanflorez219** [comment]
this pr sucks, thats how you code? say goodbye to your current job 🤢

File: telegram/commands/report/run.go (hunk 3 above)
Comment (bescobarm) [reply telegram/commands/report/run.go:29]: copy paste mistake i guess, kr -> keratineReport :v

File: telegram/commands/report/run.go (hunk 4 above)
Comment (bescobarm) [reply telegram/commands/report/run.go:67]: if we want modify the report message in the future probably the sort map won't work

File: telegram/commands/report/tooth.go (hunk 6 above)
Comment (bescobarm) [reply telegram/commands/report/tooth.go:28]: same of above jsjs wr --> water report :v it must be modified xd

File: telegram/commands/report/all.go (hunk 1 above)
Comment (valerymolinalika) [reply telegram/commands/report/all.go]: Done

File: telegram/commands/commands.go [hunk 7]
```diff
@@ -34,6 +34,10 @@ var (
 		"water":    report.SendWaterReport,
 		"keratine": report.SendKeratineReport,
 		"pipi":     report.SendPipiReport,
+		"shower":   report.SendShowerReport,
+		"run":      report.SendRunReport,
+		"tooth":    report.SendToothReport,
```
[the remark this answers is not in the document]
Comment (valerymolinalika) [reply telegram/commands/commands.go:41]: yes, tooth

File: telegram/commands/report/run.go (hunk 2 above)
Comment (valerymolinalika) [reply telegram/commands/report/run.go:21]: yo vere

File: telegram/commands/report/run.go (hunk 3 above)
Comment (valerymolinalika) [reply telegram/commands/report/run.go]: fixed

File: telegram/commands/report/shower.go [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+	for _, activity := range showerActivities {
+		createdAt, err := time.Parse(time.RFC3339, activity.CreatedAt)
+		if err != nil {
+			return "", shared.SendMessage(bot, chatID, fmt.Sprintf(shared.ErrSendMessage, err.Error()))
+		}
+
+		tookShower[createdAt.Weekday()] = labelTookShower
+	}
+
+	var currentSisasStreak, currentNonasStreak, lastNonasStreak, lastSisasStreak int
```
[the remark this answers is not in the document]
Comment (valerymolinalika) [reply telegram/commands/report/shower.go:73]: Done

File: telegram/commands/report/tooth.go (hunk 6 above)
Comment (valerymolinalika) [reply telegram/commands/report/tooth.go]: fixed

File: telegram/commands/report/shower.go (hunk 5 above)
Comment (valerymolinalika) [reply telegram/commands/report/shower.go:29]: please look at the logic, the name is correct

**Review by bescobarm — APPROVED** [review]

**Pull request merged**